pkg/api/validation/validation.go

```diff
@@ -444,6 +444,10 @@ func validateVolumeSource(source *api.VolumeSource, fldPath *field.Path) field.E
 			numVolumes++
 			allErrs = append(allErrs, validateISCSIVolumeSource(source.ISCSI, fldPath.Child("iscsi"))...)
 		}
```

Comment:
@liggitt @rootfs 
IIRC, the InitiatorName should have follow IQN (iSCSI Qualified Name) format or EUI (Extended Unique Identifier) format to connect to iSCSI target properly.
ex.
```
cat /etc/iscsi/initiatorname.iscsi
InitiatorName=iqn.1994-05.com.redhat:185ce16b55ad
```

But from my quick check, it seems that iscsiadm accepts any kind of initiatorname like below. Then iSCSI connection will fail if the initiatorname doesn't align specific format as mentioned above. 
Since iscsiadm doesn't validate initiatorname, I think we can skip validation in Kubernetes.
Also we don't validate "IQN" which is already in ISCSIVolumeSource struct.
Any thought?

```
root# sudo iscsiadm -m iface -I stable -o update -n iface.initiatorname -v a
stable updated.
root# sudo iscsiadm -m iface -I stable | grep iface.initiatorname
iface.initiatorname = a
root# sudo iscsiadm -m iface -I stable -o update -n iface.initiatorname -v $
stable updated.
root# sudo iscsiadm -m iface -I stable | grep iface.initiatorname
iface.initiatorname = $
root# sudo iscsiadm -m iface -I stable -o update -n iface.initiatorname -v @
stable updated.
root# sudo iscsiadm -m iface -I stable | grep iface.initiatorname
iface.initiatorname = @
root# sudo iscsiadm -m iface -I stable -o update -n iface.initiatorname -v !
stable updated.
root# sudo iscsiadm -m iface -I stable | grep iface.initiatorname
iface.initiatorname = !
```